src/common/transformations/src/transformations/sdpa_to_paged_attention/total_sequence_length_pattern.cpp

```diff
@@ -23,21 +23,62 @@ ov::pass::TotalSequenceLengthPattern::TotalSequenceLengthPattern(
     auto kv_current = pattern::any_input();
     auto kv_concat = pattern::wrap_type<v0::Concat>({kv_gather, kv_current});
     auto kv_shape = pattern::wrap_type<v3::ShapeOf>({kv_concat});
-    auto seq = pattern::wrap_type<v8::Gather>({kv_shape, pattern::any_input(), pattern::any_input()});
+    auto gather_idx_label = pattern::wrap_type<v0::Constant>();
+    auto seq = pattern::wrap_type<v8::Gather>({kv_shape, gather_idx_label, pattern::any_input()});
 
     ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
         // TODO: Check that seq has axis that really takes sequence len but not any other dimension --
         //  use symbolic infra or look at the constant input
-        auto gather = m.get_match_root();
-        auto target_type = gather->get_output_element_type(0);
-        std::shared_ptr<Node> replacement = max_context_len;
-        if (replacement->get_output_element_type(0) != target_type) {
-            replacement = std::make_shared<v0::Convert>(replacement, target_type);
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto concat = std::dynamic_pointer_cast<v0::Concat>(pattern_map.at(kv_concat).get_node_shared_ptr());
+        auto gather = pattern_map.at(seq).get_node_shared_ptr();
+        auto gather_idx =
+            std::dynamic_pointer_cast<v0::Constant>(pattern_map.at(gather_idx_label).get_node_shared_ptr());
+        auto gather_idx_data = gather_idx->cast_vector<int64_t>();
+
+        if (!concat || !gather || !gather_idx) {
+            return false;
+        }
+
+        if (gather_idx_data.size() == 0) {
```

Comment:
size == 0?
I believe it should be size == 1
why the new added tests passed? do we actually run them in pre-commit? could you double check?

Reply:
Checking for size not to be 0. Basically, verifying that I'm able to get the result. It might make sense to check for size to be 1 only.

Reply:
Fixed.